src/ApplicationInsights.Kubernetes/TelemetryInitializers/KubernetesTelemetryInitializer.cs

```diff
@@ -33,6 +46,7 @@ internal class KubernetesTelemetryInitializer : ITelemetryInitializer
             _logger.LogDebug(@"Initialize Application Insights for Kubernetes telemetry initializer with Options:
 {0}", JsonConvert.SerializeObject(_options));
 
+            _telemetryKeyCache = telemetryKeyCache ?? throw new ArgumentNullException(nameof(telemetryKeyCache));
```

Comment:
Or use an instance of the (standard) `TelemetryCache` if nothing is passed to the constructor

Reply:
Are you expecting something like this?
```csharp
_telemetryKeyCache = telemetryKeyCache ?? TelemetryKeyCache.Default;
```

If that is the case: no. :-) In sprite of making the dependency explicit, ITelemetryKeyCache is required, not nullable.

Thinking on this is: I wouldn't mind to provide an implementation of `DefaultTelemetryKeyCache` if that interface is exposed (public). But since this is a requirement for our own code (internals), I prefer to keep it this way.

Feel free to start up a discussion offline with me :-)